src/types/StableTopUp.sol

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0 <0.9.0;
+
+import {IERC20} from "@openzeppelin/interfaces/IERC20.sol";
+
+import "../BaseConditionalOrder.sol";
+import {ConditionalOrdersUtilsLib as Utils} from "./ConditionalOrdersUtilsLib.sol";
+
+// --- error strings
+
+/// @dev The buy token balance is above the (minimum) threshold.
+string constant SUFFICIENT_BALANCE = "buyToken balance above threshold";
+
+/**
+ * @title A smart contract that tops up the buyToken balance to a specified topUpValue whenever the balance falls below threshold.
+ */
+contract StableTopUp is BaseConditionalOrder {
+    struct Data {
+        IERC20 sellToken;
+        IERC20 buyToken;
+        address receiver;
+        uint32 validityBucketSeconds;
+        uint256 threshold;
+        uint256 topUpTo;
+        bytes32 appData;
+    }
+
+    /**
+     * @inheritdoc IConditionalOrderGenerator
+     * @dev If the `receivers`'s balance of `buyToken` is below the specified threshold, tops up the receiver to `topUpTo`
+     * with from sellToken at the current market price (no limit!).
+     */
+    function getTradeableOrder(
+        address,
+        address,
+        bytes32,
+        bytes calldata staticInput,
+        bytes calldata
+    ) public view override returns (GPv2Order.Data memory order) {
+        /// @dev Decode the payload into the trade below threshold parameters.
+        StableTopUp.Data memory data = abi.decode(staticInput, (Data));
+
+        uint256 balance = data.buyToken.balanceOf(data.receiver);
+        // Don't allow the order to be placed if the balance is less than the threshold.
+        if (balance >= data.threshold) {
+            revert IConditionalOrder.OrderNotValid(SUFFICIENT_BALANCE);
```

Comment:
Note that you don't want to use `OrderNotValid`, you should actually use a Poll reversion hint so you can indicate to the watch-tower when to poll again next. `OrderNotValid` is used for when the `staticInput` doesn't conform to the conditional order type that you've written (for example, you may say that the threshold cannot be 0, as otherwise it's always valid, and it would create invalid orders, in which case you would fire `OrderNotValid` as a revert reason).

Reply:
Would it be reasonable to set a poll frequency? Like check the balance every day or every hour? Would be super cool if there was some way to indicate "watch this account for outgoing transfers" but I'm not sure if it fits within this setup...

Reply:
Nice docs! I guess the most appropriate thing (although it's a lot) would be PollTryNextBlock or next epoch.

Reply:
You can only indicate when the watch-tower should _retry_ unfortunately, not when it should poll next given a valid polling (this is something I'd like to address in a new iteration)